### ISSUE TITLE
Tab index on grid seems to break it

### ISSUE THREAD

**Issue by stefanuebe** [vaadin/flow-components#2181]
### Description
I have a simple grid with a tab index. When now focusing it using "focus()" or tabbing through the page, it focus the grid at some point. When the grid itself is focused (not the cells), it breaks totally, when using the keyboard to navigate further (arrows and tab)

### Expected outcome
It should not break.

### Actual outcome
Grid has the "focus" after tabbing through or using "focus()" on the grid element.
![image](https://user-images.githubusercontent.com/36725474/135809899-087c946c-f934-4e29-a3f6-c3024b638ab3.png)

Using tab key after focusing the grid
![image](https://user-images.githubusercontent.com/36725474/135809695-38a8a0ab-41be-47a1-8f6e-36afdd303a2e.png)

Using arrow keys after focusing the grid
![image](https://user-images.githubusercontent.com/36725474/135809793-9850a5fa-6ae5-4054-a090-7391af93493e.png)

### Minimal reproducible example

```
@Route(value = "grid-focus-index-issue", layout = MainLayout.class)
public class GridFocusIssueView extends Div {

    public GridFocusIssueView() {
        List<SamplePerson> persons = new ArrayList<>();

        persons.add(new SamplePerson("Henry"));
        SamplePerson itemToSelect = new SamplePerson("Indiana");
        persons.add(itemToSelect);
        persons.add(new SamplePerson("Jones"));

        Grid<SamplePerson> grid = new Grid<>();
        grid.addColumn(SamplePerson::getName);

        grid.setItems(persons);
        grid.addFocusListener(event -> System.out.println("focus grid"));

        grid.setTabIndex(1); // this one here breaks everything!

        TextField textField = new TextField("", "Arrow down here to focus grid");
        textField.setWidth("300px");
        Shortcuts.addShortcutListener(textField, grid::focus, Key.ARROW_DOWN);

        add(new Button("Focus", event -> grid.focus()), textField, grid);
    }

    public static class SamplePerson {
        private final String name;

        public SamplePerson(String name) {
            this.name = name;
        }

        public String getName() {
            return name;
        }
    }
}
```

### Steps to reproduce

1. Open the page
2a. Focus the grid by using either the tab key and tab through the page until the grid is outlined with a black line
2b. Use the Focus button (the black outline will not appear here immediately)
2c. Focus the text field and use then arrow down key
3. Use the tab key or the up/down arrows.
4. Errors should appear 

### Environment

- Version of the component / Vaadin platform: Vaadin 14
- OS: Windows 10
- Tested in Chrome 93

**Comment by TatuLund** [comment]
Yes, this happens because Grid really do not implement focus properly, see also https://github.com/vaadin/flow-components/issues/1254

**Comment by TatuLund** [comment]
Also, as now from Vaadin 21, there are cell based focus tracking in Grid, it would be clearer that focusing Grid would mean focusing the Grid's first cell instead of the Grid. Thus setting tab index to Grid should work accordingly, i.e. delegate the focus to the first cell.
